pydantic/class_validators.py

```diff
@@ -91,29 +94,36 @@ def dec(f: AnyCallable) -> classmethod:
 
 
 def root_validator(
-    _func: Optional[AnyCallable] = None, *, pre: bool = False
+    _func: Optional[AnyCallable] = None, *, pre: bool = False, allow_reuse: bool = False
 ) -> Union[classmethod, Callable[[AnyCallable], classmethod]]:
     if _func:
-        _check_validator_name(_func)
-        f_cls = classmethod(_func)
+        f_cls = classmethod(_func) if not isinstance(_func, classmethod) else _func
+        _func = f_cls.__func__
+        _check_validator_name(_func, allow_reuse)
         setattr(f_cls, ROOT_VALIDATOR_CONFIG_KEY, Validator(func=_func, pre=pre))
         return f_cls
 
     def dec(f: AnyCallable) -> classmethod:
-        _check_validator_name(f)
-        f_cls = classmethod(f)
+        f_cls = classmethod(f) if not isinstance(f, classmethod) else f
+        f = f_cls.__func__
+        _check_validator_name(f, allow_reuse)
         setattr(f_cls, ROOT_VALIDATOR_CONFIG_KEY, Validator(func=f, pre=pre))
         return f_cls
 
     return dec
 
 
-def _check_validator_name(f: AnyCallable) -> None:
+def _check_validator_name(f: AnyCallable, allow_reuse: bool) -> None:
     """
     avoid validators with duplicated names since without this, validators can be overwritten silently
     which generally isn't the intended behaviour, don't run in ipython - see #312
+
+    does not check for duplicated names in functions defined at the module-global level
+    since such validators are likely intended for reuse
     """
-    if not in_ipython():  # pragma: no branch
+    if in_ipython():  # pragma: no cover
+        return
+    if not allow_reuse and '.' in f.__qualname__:
```

Comment:
This could be confusing. We need to explicitly document this behaviour.

Reply:
@samuelcolvin Should we just remove this and explicitly require the validator to set `allow_reuse=True`, even at the global level? Seems like it would be easier to document and ultimately more intuitive.